src/qibolab/_core/parameters.py

```diff
@@ -202,3 +218,118 @@ def replace(self, update: Update) -> "Parameters":
             _setvalue(d, path, val)
 
         return self.model_validate(d)
+
+
+QubitMap = dict[QubitId, Qubit]
+InstrumentMap = dict[InstrumentId, Instrument]
+
+
+class Hardware(TypedDict):
```

Comment:
Changed to `Model` in 1d0ad86c7c13090f2b582c28dafce6c0af4fdff5 (and also e409ea60329f2c1ccc357e0a85e71eb81988571a fixing `create_platform`). I guess the major difference from a usability point of view is that now if someone wants to write a `create` method that returns `Hardware` instead of `Platform` they have to import it and define it properly, while before they could just return a `dict` (even though formally, if using annotations, they would still have to import the type).